ztp/siteconfig-generator/siteConfig/siteConfigHelper.go

```diff
@@ -173,3 +174,28 @@ func deleteInspectAnnotation(bmhCR map[string]interface{}) map[string]interface{
 	}
 	return bmhCR
 }
+
+// buildJsonAnnotation returns string in json format
+func buildJsonAnnotation(networkType, configOverrides string) (string, error) {
+
+	networkAnnotation := "{\"networking\":{\"networkType\":\"" + networkType + "\"}}"
+	if !json.Valid([]byte(networkAnnotation)) {
+		return "", fmt.Errorf("Invalid json conversion of network type")
+	}
+
+	if configOverrides != "" {
+		if !json.Valid([]byte(configOverrides)) {
+			return "", fmt.Errorf("Invalid json parameter set at installConfigOverride")
+		}
+		trimmedConfigOverrides := strings.TrimPrefix(configOverrides, "{")
+		trimmedNetworkType := strings.TrimSuffix(networkAnnotation, "}")
+		finalJson := trimmedNetworkType + "," + trimmedConfigOverrides
```

Comment:
What will happen if the "network" is also included in the installConfigOverrides, perhaps with other non-overlapping networking settings?

Reply:
For example, if we have below in siteconfig:
```
networkType: OVNKubernetes
installConfigOverrides "{\"networking\":{\"UserManagedNetworking\":\"True\"}}"`
```
Do we want the annotation to be for example in array format for the networking:

```
{"networking":[{"networkType":"OVNKubernetes"},{"UserManagedNetworking":"True"}]}
```

Reply:
The content of this JSON string (the entire overrides) feeds into the [installConfig defined here](https://github.com/openshift/installer/blob/53727e66b810286f5f10441470623f0a8bb11b1c/pkg/types/installconfig.go#L87) (note the Network on line 115). The Network isn't a list/array, so I don't think we can embed it as a list entry. If the user gives us a networkType I think you can take what is in your array and do a "dictionary merge" with any other networking content.

Reply:
Fixed with commit 0843a14efa1eb6c8e7d476a1d7197a790758d76f, please look at the tests that are added.